docs/content/preview/architecture/transactions/concurrency-control.md

```diff
@@ -0,0 +1,1166 @@
+or aborted by a conflict: 40001
+```
+
+```sql
+rollback;
+```
+
+   </td>
+  </tr>
+</tbody>
+</table>
+
+### Die example
+<table class="no-alter-colors">
+<thead>
+    <tr>
+    <th>
+    Client 1
+    </th>
+    <th>
+    Client 2
+    </th>
+    </tr>
+  </thead>
+  <tbody>
+  <tr>
+   <td>
+   </td>
+   <td>
+
+```sql
+SET yb_transaction_priority_lower_bound = 0.6;
+```
+
+   </td>
+  </tr>
+  <tr>
+   <td>
+
+```sql
+SET yb_transaction_priority_upper_bound = 0.4;
+```
+
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>
+   </td>
+   <td>
+
+```sql
+begin transaction isolation level repeatable read;   
+```
+
+   </td>
+  </tr>
+  <tr>
+   <td>
+   </td>
+   <td>
+
+```sql
+select * from test where k=1 for update;
+```
+
+```output
+ k | v
+---+---
+ 1 | 1
+(1 row)
+```
+   </td>
+  </tr>
+  <tr>
+   <td>
+
+```sql
+begin transaction isolation level repeatable read;   
+```
+
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>
+
+```sql
+select * from test where k=1 for update;
+```
+
+```output
+ERROR:  All transparent retries exhausted. could not serialize
+access due to concurrent update
+```
+
+```sql
+rollback;
+```
+
+   </td>
+   <td>
+   </td>
+  </tr>
+</tbody>
+</table>
+
+### Best-effort internal retries for first statement in a transaction
+
+Note in the above example that the error message says `All transparent retries exhausted`. This is because, if the transaction T1, while exeuting the first statement, finds other concurrent conflicting transactions with equal or higher priority, then T1 will perform a few retries with exponential backoff before giving up in anticipation that the other transactions will be done in some time. The number of retries are configurable by the `ysql_max_write_restart_attempts` TServer gflag and the exponential backoff parameters are the same as the ones described [here](../read-committed/#performance-tuning).
+
+Each retry will work a newer snapshot of the database in anticipation that the conflicts might not occur (if an earlier conflicting transaction T2 has committed with a commit time before the read time of the new snapshot, the conflicts with T2 would essentially be voided since T1 and T2 would no longer be "concurrent").
+
+Also note that the retries will not be performed in case the amount of data to be sent from YSQL to the client proxy exceeds the TServer gflag `ysql_output_buffer_size`.
+
+## Wait-on-Conflict [[Beta]](/preview/faq/general/#what-is-the-definition-of-the-beta-feature-tag)
+
+This mode of concurrency control is applicable only for YSQL and provides the same semantics as PostgreSQL.
+
+In this mode, transactions are not assigned priorities. If a conflict occurs when a transaction T1 tries to read/ write/ lock a row in a conflicting mode with a few other concurrent transactions, T1 will **wait** until all conflicting transactions finish by either committing or rolling back. Once all conflicting transactions have finished, T1 will:
```

Comment:
Changed to `read, write or lock`

Also, since there are just two instances of `read, write, or lock`, avoiding defining access for now. Let me know if I should still define and use `access` instead